src/components/Suggestion/LinkPicker/suggestions.js

```diff
@@ -24,9 +24,48 @@ import LinkPickerList from './LinkPickerList.vue'
 
 import { searchProvider, getLinkWithPicker } from '@nextcloud/vue/dist/Components/NcRichText.js'
 
+import menuEntries from './../../Menu/entries.js'
+
+const suggestGroupFormat = t('text', 'Formatting')
+const suggestGroupPicker = t('text', 'Smart picker')
+
+const filterOut = (e) => {
+	return ['undo', 'redo', 'outline', 'emoji-picker'].indexOf(e.key) > -1
+}
+
+const important = ['task-list', 'table']
+
+const sortImportantFirst = (list) => {
+	return [
+		...list.filter(e => important.indexOf(e.key) > -1),
+		...list.filter(e => important.indexOf(e.key) === -1),
+	]
+}
+
+const formattingSuggestions = (query) => {
+	return sortImportantFirst(
+		[
+			...menuEntries.find(e => e.key === 'headings').children,
+			...menuEntries.filter(e => e.action && !filterOut(e)),
+			...menuEntries.find(e => e.key === 'callouts').children,
+			{
+				...menuEntries.find(e => e.key === 'emoji-picker'),
+				action: (command) => command.insertContent(':'),
+			},
+		].filter(e => e?.label?.toLowerCase?.()?.includes(query.toLowerCase()))
+			.map(e => ({ ...e, suggestGroup: suggestGroupFormat })),
+	)
+}
+
 export default () => createSuggestions({
 	listComponent: LinkPickerList,
 	command: ({ editor, range, props }) => {
+		if (props.action) {
+			const commandChain = editor.chain().deleteRange(range)
```

Comment:
The `deleteRange` is so this only creates new elements, right? I cannot use it to format the current selection?
(I'll try out - just reviewing the code first)

Reply:
The range contains the character(s) used to trigger the suggestions `/`.